utils/markdown.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package utils
+
+import (
+	"strings"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/extension"
+	astExt "github.com/yuin/goldmark/extension/ast"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/util"
+
+	"github.com/mattermost/mattermost-server/v5/shared/mlog"
+)
+
+func StripMarkdown(markdown string) string {
+	md := goldmark.New(
+		goldmark.WithExtensions(extension.GFM),
+		goldmark.WithRenderer(
+			renderer.NewRenderer(renderer.WithNodeRenderers(
+				util.Prioritized(newNotificationRenderer(), 500),
+			)),
+		),
+	)
+
+	var buf strings.Builder
+	if err := md.Convert([]byte(markdown), &buf); err != nil {
+		mlog.Warn("failed parse to markdown")
+		return markdown
+	}
```

Comment:
The idea is to return the error to the caller. Like `StripMarkdown(string) (string, error)`. And then in `notification_push.go`, check the error, and log it, and continue with the original string.

This is a low level function, and should not be deciding how to respond to an error.

Reply:
I thought you want me to just logged it, sorry I will update.